### ISSUE TITLE
Horse Name/Title

### ISSUE THREAD

**Issue by inerdy** [decaprime/LeadAHorseToWater#1]
When you over your cursor over the horse instead of the name color changing you just see html info.
![image](https://user-images.githubusercontent.com/11263146/175739299-bf4e5d76-22e5-4375-87ca-d84603d8e5a6.png)

Then left the server up for a few hours (still up) and the horse names changed 
![image](https://user-images.githubusercontent.com/11263146/175761919-5962ffb3-2a2c-47ca-a9f2-4ef17d716dac.png)

**Comment by decaprime** [comment]
I'm still unsure exactly how the second picture happened so I'm leaving this issue open. For either one though I've added toggles to let you continue to run the plugin without the rename or color if that's annoying.

**Comment by firelook** [comment]
I have the same problem : perhaps you save the name with drinking then when you reloead you re add drinking ;)

**Comment by decaprime** [comment]
Thanks @firelook  , take a look for yourself [here](https://github.com/decaprime/LeadAHorseToWater/blob/main/Plugin.cs#L178) if your interested but the code checks for the prefix and operates based on if it has it or not, not on save/load or anything.

I've never been able to reproduce this bug. Can you please include some steps to reproduce this issue so I can fix it? If not, could you please detail things like what you were doing when this occurred, was it only one horse, name of the horse before, info about the server, etc. All that info can help narrow down why this happened.

**Comment by firelook** [comment]
thanks, but perhaps you should do a find instead of a starting (because we have </color => means we have a miss starting hidden specific characters => a guess )

To test I have done : EnablePrefixColor=false 
https://i.imgur.com/kjP4mfH.jpeg
and it is worst
perhaps it will help you

**Comment by decaprime** [comment]
I think you're continuing to test with the same horse and haven't reset it's name, so I'm not seeing that as a reproduction unless I misunderstand.

I suspect that the first time this happens, the prefix is there, but it fails to match. Once you hit the name limit it's going to continuously happen as the entire prefix doesn't fit in there and it continues not to match; however I know this happens with the default name, so it's not only on maximum length horse names. I am not out of things to try, but without a way of testing them, I can't properly approach this.

**Comment by firelook** [comment]
if I understand you test to check if the name start with the prefix drinking, but if you put colors your test failed no ? you can put either colors, drinking or both, => are you sure you test all combinations please ?

P.S.
As suggested I change the type of horse and now seems ok

**Comment by decaprime** [comment]
Yes I tested prefix with and without colors but I have not reproduced this issue. The code results in the prefix you currently have configured with the color/formatting, what comes out is a string it uses to check for the prefix. [https://github.com/decaprime/LeadAHorseToWater/blob/main/Plugin.cs#L175](https://github.com/decaprime/LeadAHorseToWater/blob/main/Plugin.cs#L175), (if you're unfamiliar with how you can read about [`? :`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/conditional-operator)) 

If you have a horse that has the prefix and then you turn the server off, change the config to be a different prefix (in any way), and start it up again the horse is going to have a mangled name because it doesn't know the history what you previously used, please note that's to be expected, and probably not this bug. This bug is the one in the OP: `Then left the server up for a few hours (still up) and the horse names changed`.  

I'll try to explain the code more: every 1.5s it's processing an 'update' which gets the horses in range of the wells, then for each of those horses it runs that [update name code](https://github.com/decaprime/LeadAHorseToWater/blob/main/Plugin.cs#L178), 99% of the calls immediately exit and do nothing because it already has the prefix, is already drinking, or is already not-drinking . I've tried really fast, I've tried after a long wait, I've tried with multiple horses, multiple wells, moving wells between horses already drinking, I've *tried* simulating lag, and a lot more. My current guess is it is some type of lag/sync issue but I haven't made it happen.

edit to add: I have some plans to keep a database of horses/horse data for *other* reasons, which would fix this bug eventually as we'd use the database name and write the prefix with it, and not be storing it back with the prefix leading to this looping.

**Comment by firelook** [comment]
I try to reproduce the problem, and I do not manage
but anyway I keep you plugins and like it 
P.S.
perhaps you can extend your plugins to take into account also other water like imperious large fountain ;)
